src/components/Header.tsx

```diff
@@ -0,0 +1,123 @@
+import { Dispatch, FC, FormEvent, useEffect, useState } from 'react';
+import { USER_ID, addTodo, updateTodos } from '../api/todos';
+
+import useFocusInput from '../hooks/useFocusInput';
+import { Todo } from '../types/Todo';
+
+interface Props {
+  onErrorMessage: (message: string) => void;
+  setTodos: Dispatch<React.SetStateAction<Todo[]>>;
+  setTempTodo: Dispatch<React.SetStateAction<Todo | null>>;
```

Comment:
Don't expose more than you need 
```suggestion
  setTempTodo: (todo: Todo | null) => void;
```